### ISSUE TITLE
Diffusion model when train the whole network

### ISSUE THREAD

**Issue by rdf87** [NUS-HPC-AI-Lab/Neural-Network-Diffusion#20]
Hi~ 
I plan to use MNIST to reconstruct the parameters of LeNet5. 
In the first 30000 epochs of training the autoencoder, I achieved good accuracy.
But in the last 30000 epochs of training the diffusion model, the accuracy plummeted to 10%.
Then I tried to generate the weights only of the layer fc3 in the LeNet5 network, which can improve the accuracy to 90%.
That is to say, I cannot reproduce the experiment of generating the whole network in your article. In addition, according to the data in Table 3 of the article, the number of parameters of the LeNet5 network should be within the generation capability range of the diffusion model.
May I ask which generation model you used for the last 30000 epochs when generating the whole network? I tried all the generation models provided in your code, but did not get good results.

**Comment by MTDoven** [comment]
Hi, we are so sorry to have put you to such trouble. Here are some possible solutions.

1. Check the value of your latent encoded by autoencoder. If AE overfits, which is a normal phenomenon, sometimes the value of your latent will get very large, for example around 100. In this case, we can manually scale the value to around (-0.5, 0.5), which will make diffusion training easier. (Here is another suggested operation, which can remove the "tanh" of the last layer of the upsampling of U-Net)

2. Check the parameters of each layer to see if the mean and variance are too different from each other. For example, the values in the first layer are around 10.0 and the values in the second layer are around 0.01. It doesn't matter if the mean and variance are slightly different. But if the differences are too large, you may need to manually normalize them.

3. Check your "latent_dim": Usually, the "latent_dim" needs to be adjusted based on the number of parameters. Specifically, for your experiment, LeNet with 10714 parameters, "latent_dim" can be 64 or 128 from our experience. Check if your "latent_dim" is set too small.

4. If all above doesn't work, you can try making the depth of U-Net in diffusion a little deeper and using the kernel size 3. This will make diffusion easier to train. And you can also try tuning the other hyperparameters.

Good luck!
